internal/rbd/rbd_journal.go

```diff
@@ -521,28 +521,45 @@ func undoVolReservation(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
 // complete omap mapping between imageName and volumeID.
 
 // RegenerateJournal performs below operations
+// Extract required parameters journalPool, pool from volumeAttributes
+// Extract optional parameter volumeNamePrefix from volumeAttributes
 // Extract information from volumeID
 // Get pool ID from pool name
 // Extract uuid from volumeID
 // Reserve omap data
 // Generate new volume Handler
 // The volume handler won't remain same as its contains poolID,clusterID etc
 // which are not same across clusters.
+// nolint:gocyclo,cyclop // TODO: reduce complexity
 func RegenerateJournal(
-	imageName, volumeID, pool, journalPool, requestName string,
+	volumeAttributes map[string]string,
+	volumeID, requestName string,
 	cr *util.Credentials) (string, error) {
 	ctx := context.Background()
 	var (
-		options map[string]string
-		vi      util.CSIIdentifier
-		rbdVol  *rbdVolume
+		options           map[string]string
+		vi                util.CSIIdentifier
+		rbdVol            *rbdVolume
+		pool, journalPool string
+		ok                bool
 	)
 
+	if pool, ok = volumeAttributes["pool"]; !ok {
+		return "", fmt.Errorf("required 'pool' parameter missing in volume attributes")
+	}
+	if journalPool, ok = volumeAttributes["journalPool"]; !ok {
+		return "", fmt.Errorf("required 'journalPool' parameter missing in volume attributes")
+	}
+	kmsID, err := util.FetchEncryptionKMSID(volumeAttributes)
```

Comment:
instead of blindly calling `fetchEn..` , just check for `volumeattaributes-> encrypted` key and call that function. Also you dont need entire volumeattritbutes inside the Fetch, rather just pass values of `encrypted and ..kmsid` , I would rather functionalize the `bool` parsing to a seperate function something like `isValidBool()` which can be generic ( can be used for other value parsing of bools) and can be used for parsing `encrypted` value in this case.